### DISCUSSION TITLE
OAuth login to specific organization

### DISCUSSION THREAD

**Discussion by ocherfas** [grafana/grafana#39101] [General]
Is it possible to configure Grafana OAuth to be applied to a specific organization?

For example I want to configure generic OAuth with Okta, that when user uses it, grafana will log the user to preconfigured organization that is not the default one.

I tried using allowed_organizations configuration key but I figured it is used for configuration received from the OAuth provider rather than Grafana organizations.

Thanks!

**Comment by torkelo** [comment]
No this is currently not possible

**Comment by RedHat93** [comment]
Ciao guys,

is there any update regarding this ticket?
We have the same need, we would like to configure OAuth Access into our Grafana.

But at the same time we would like to assign specific role (Editor/Viewer) for the different Organizations we've configured.

If this is still unsupported, have you got any workaround to make our needs work?

Thanks you so much :)

**Comment by fatalglitch** [comment]
The other request like this was closed per the above, so is there any update on roadmap for this feature request? Assigning orgs via OAuth is a pretty important feature if you are using Organizations.

**Comment by RyanW8** [comment]
This would be a great feature to have. It would be even better if we could map oAuth groups or any other attributes to specific orgs.

**Comment by AntoineMicheaOpencell** [comment]
Agree, that could be very useful to be able to create multi-tenant dashboards.

**Comment by Fufusulier** [comment]
Indeed, very useful

**Comment by m-yosefpor** [comment]
any updates on this?

**Comment by kminehart** [comment]
Since this is an old feature request with some duplicate issues, I'm going to convert this to a discussion where we can continue discussing this.

**Comment by nicolas17** [comment]
It's especially weird that the LDAP authentication module does support mapping LDAP groups to org_id + role in that org, but the OAuth module doesn't...

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82992. Feel free to continue the discussion around this there. Thank you!
